### PR TITLE
Add copy function to support storage migration task

### PR DESCRIPTION
**Pull request by chaoyli**

### CODE REVIEW

File: be/src/olap/rowset/alpha_rowset.cpp [hunk 1]
```diff
@@ -219,6 +219,20 @@ OLAPStatus AlphaRowset::make_snapshot(const std::string& snapshot_path,
     return OLAP_SUCCESS;
 }
                                     
+OLAPStatus AlphaRowset::copy_files_to_path(const std::string& dest_path,
+                                           std::vector<std::string>* success_files) {
```
Comment (yiguolei) [inline be/src/olap/rowset/alpha_rowset.cpp]: should add copy_files_to_path to parent class

File: be/src/olap/rowset/segment_group.cpp [hunk 2]
```diff
@@ -723,6 +723,39 @@ OLAPStatus SegmentGroup::make_snapshot(const std::string& snapshot_path,
...
 }
 
+OLAPStatus SegmentGroup::copy_files_to_path(const std::string& dest_path,
+                                            std::vector<std::string>* success_files) {
+    if (_empty) {
+        return OLAP_SUCCESS;
+    }
+    for (int segment_id = 0; segment_id < _num_segments; segment_id++) {
+        std::string dest_data_file = construct_data_file_path(dest_path, segment_id);
+        if (!check_dir_existed(dest_data_file)) {
```
Comment (yiguolei) [inline be/src/olap/rowset/segment_group.cpp]: If exist file, then should check its length  or remove the exist file?

File: be/src/olap/rowset/alpha_rowset.cpp (hunk 1 above)
Comment (chaoyli) [reply be/src/olap/rowset/alpha_rowset.cpp]: Already added.

File: be/src/olap/task/engine_storage_migration_task.cpp [hunk 3]
```diff
@@ -219,12 +219,16 @@ OLAPStatus EngineStorageMigrationTask::_copy_index_and_data_files(
...
         std::vector<RowsetSharedPtr>& consistent_rowsets) {
-    // TODO(lcy). copy function should be implemented
+    std::vector<std::string> success_files;
+    OLAPStatus status = OLAP_SUCCESS;
     for (auto& rs : consistent_rowsets) {
-        std::vector<std::string> success_files;
-        RETURN_NOT_OK(rs->make_snapshot(schema_hash_path, &success_files));
+        status = rs->copy_files_to_path(schema_hash_path, &success_files);
+        if (status != OLAP_SUCCESS) {
+            while (OLAP_SUCCESS != remove_all_dir(schema_hash_path));
```
Comment (imay) [inline be/src/olap/task/engine_storage_migration_task.cpp]: Don't dead loop here

File: be/src/olap/rowset/segment_group.cpp (hunk 2 above)
Comment (chaoyli) [reply be/src/olap/rowset/segment_group.cpp]: I check dir exist or not firstly.

File: be/src/olap/task/engine_storage_migration_task.cpp (hunk 3 above)
Comment (chaoyli) [reply be/src/olap/task/engine_storage_migration_task.cpp]: I will fatal in this place.

File: be/src/olap/rowset/alpha_rowset.h [hunk 4]
```diff
@@ -85,7 +85,9 @@ class AlphaRowset : public Rowset {
     int64_t ref_count() const override;
 
     OLAPStatus make_snapshot(const std::string& snapshot_path,
-                             std::vector<std::string>* success_files) override;
+                             std::vector<std::string>* success_links) override;
+    OLAPStatus copy_files_to_path(const std::string& dest_path,
+                                  std::vector<std::string>* success_files);
```
Comment (kangpinghuang) [inline be/src/olap/rowset/alpha_rowset.h]:
```suggestion
                                  std::vector<std::string>* success_files) override;
```

**Review by kangpinghuang — CHANGES_REQUESTED** [review]

**Review by yiguolei — APPROVED** [review]

**Pull request merged**